ios/engine/KMEI/KeymanEngine/Classes/Keyboard/InputViewController.swift

```diff
@@ -347,18 +401,11 @@ open class InputViewController: UIInputViewController, KeymanWebDelegate {
       perform(#selector(self.enableInputClickSound), with: nil, afterDelay: 0.1)
     }
 
-    var hasDeletedSelection = false
+    // `true` if there was selected text to be deleted
+    let deletedSelection = self.deleteSelection()
 
-    if let selected = textDocumentProxy.selectedText {
-      if selected.count > 0 {
-        textDocumentProxy.deleteBackward()
-        hasDeletedSelection = true
-      }
-    }
-
-    if numCharsToDelete <= 0 || hasDeletedSelection {
+    if numCharsToDelete <= 0 || deletedSelection {
       textDocumentProxy.insertText(newText)
```

Comment:
This seems that if (a) we have no selection, and (b) no deletion, and (c) text to insert, then it won't insert the new text?

Reply:
No, it just comes later, after deleting preceding text.  In the case that no such deletions are needed, this section allows us to return early before all of the delete-left handling kicks in.  No edits were made near the later code block, so it doesn't show up in the default code-review code-windows.

Reply:
But... I think there is something off here; we probably shouldn't condition on the `deletedSelection` bit.

Reply:
Oh right.  If we do have a selection, we currently act as if the context is nil and thus can't back delete, so it's probably a moot point.

Reply:
The "enforcer" for "selected text" = "empty context", at least within the internal Web engine:

https://github.com/keymanapp/keyman/blob/cffabd719edf6c32297120e2a1ab5639ddc50faa/common/web/keyboard-processor/src/text/kbdInterface.ts#L302-L306